src/app/services/pokemon.service.ts

```diff
@@ -15,10 +18,30 @@ export class PokemonService {
    * Get pokemon by name
    * 
    * @param pokemonName pokemon name to search for
-   * @returns { Observable<any> }
+   * @returns { Observable<Pokemon> }
    */
-  getPokemon(pokemonName: string) : Observable<any> {
-    return this.http.get(`${APIURL}${pokemonName}`)
+  getPokemon(pokemonName: string) : Observable<Pokemon> {
+    const response = this.http.get<any>(`${APIURL}${pokemonName}`);
+    return response.pipe( map((data: any) => {
+      return {
+        id: data.id,
+        name: data.name,
+        sprite: data.sprites.front_default,
+        animateSprite: data.sprites.versions['generation-v']['black-white'].animated.front_default,
+        types: data.types.map((type: any) => type.type.name),
+        height: data.height,
+        weight: data.weight,
+        abilities: data.abilities.map((ability: any) => ability.ability.name),
+        stats: data.stats.map((stat: any) => {
+          return {
+            name: stat.stat.name,
+            value: stat.base_stat
+          }
+        }),
```

Comment:
meglio i return compatti
```suggestion
        stats: data.stats.map((stat: any) =>
          ({
            name: stat.stat.name,
            value: stat.base_stat
          })
        ),
```